### PR TITLE
airbyte-ci: trigger test on doc change

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/8960
A connector doc modification did not trigger connector tests.
This has to change now that we have tests on docs (in QA checks).

All connector tests will be triggered on doc change. If this turns out to be annoying we can change the control flow to skip code tests if only docs were modified.

@clnoll this implementation is slighlty different from what we groomed together. Please check out this [comment](https://github.com/airbytehq/airbyte-internal-issues/issues/8960) for details.

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#42046** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42046?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/connectors_tests.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
           # Note: expressions within a filter are OR'ed
           filters: |
             connectors:
-              - '*'
```
Comment (clnoll) [inline .github/workflows/connectors_tests.yml:41]: Looks like this was causing tests to run for all changes?

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 MANIFEST_FILE_NAME = "manifest.yaml"
 METADATA_ICON_FILE_NAME = "icon.svg"
 DIFF_FILTER = "MADRT"  # Modified, Added, Deleted, Renamed, Type changed
-IGNORED_FILE_EXTENSIONS = [".md"]
+IGNORED_FILE_EXTENSIONS = []
```
Comment (clnoll) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:35]: Is there any reason to keep this now that it's empty?

**Review by clnoll — APPROVED** [review]
LGTM, just a minor non-blocking comment.

**Pull request merged**